src/routine.cpp

```diff
@@ -24,6 +24,7 @@ namespace clblast {
 // The constructor does all heavy work, errors are returned as exceptions
 Routine::Routine(Queue &queue, EventPointer event, const std::string &name,
                  const std::vector<std::string> &routines, const Precision precision,
+                 std::initializer_list<const char *> source_pct,
```

Comment:
So now we have two routine modes (either the built-in host code or the custom host code) and three kernel modes (either the built-in OpenCL kernel, or the built-in pad/copy/transpose routines + custom main kernel, or fully custom kernel).